NarrowTable.lua

```diff
@@ -0,0 +1,40 @@
+local NarrowTable, parent = torch.class('nn.NarrowTable', 'nn.Module')
+
+function NarrowTable:__init(offset, length)
+   parent.__init(self)
+   self.offset = offset
+   self.length = length or 1
+   if not offset then
+      error('nn.NarrowTable(offset, length)')
+   end
+   
+   self.output = {}
+   self.gradInput = {}
+end
+
+function NarrowTable:updateOutput(input)
+   for k,v in ipairs(self.output) do self.output[k] = v end
```

Comment:
hmm, I am very confused. You are just reassigning all the integer keys of self.output to their values. I dont see what has changed after this line, self.output is exactly the same.